### PR TITLE
Updated doc regarding recent prisma changes.

### LINKED ISSUE #430: Update Documentation/README
## Description

We've had a few significant structural changes in the codebase in quite quick succession and so the docs have become out-of-date. In that time, we've had new contributors join the project and it's been challenging to get started with the outdated docs.

Closes #426

### Changes

- **j-db-client:** we moved all of our Prisma code out into an internal package so that we could import that into various micro-services (such as J-Mail for emailing) that need to interact with the database.
- **j-mail:** emails are now handled by a serverless micro-service that is all a part of our AWS cloud infrastructure, which consists of a managed queue with (SQS) and Lambda. This is currently being further updated in #417 as we build out email batching.
- **.env:** I think our instructions around the `.env` file have always been a little confusing. You should be able to have a `.env` at the root level of the project, but depending on how you're running things it can be helpful to have one in `j-db-client` as well. We could update the example files

### Other notes

- The workflow with `j-db-client` is that we actually have scripts that can be run from the top-level project to make things less confusing now, so to generate or run migrations developers can still just run `npm run migrate:save` or `npm run migrate:up` from the top level
  - To make DB changes, first edit the `schema.prisma` file as needed
  - Then, in the project root, run `npm i @journaly/j-db-client` which will re-install the library and will also run the post-install hook that generates the Prisma Client

### PR DESCRIPTION
**Pull request by JoshuaBelden**
## Update to `README.md`

Request to update the `README.md` documentation to reflect recent changes to prisma implementation. Current language references the `prisma` directory as being directly in root, but it is actually under `j-db-client`. The instructions around the `.env` file were slightly confusing, so I cleaned those up as well to be a bit more straight forward.

Details of the issue mention being able to run commands from root as well as the `prisma` folder (which actually need to be run from `j-db-client`). While some of the npm scripts can run from root, I found `npm run migrate:up` had to be specifically run from the `j-db-client` folder. It would be nice to eliminate the need for the duplicated environment file, but I'm new to a lot of this and not quite sure how to make that work.

Documentation is at least correct for the current state of the repo.

**Issue:** closes #430

...

https://www.journaly.com/dashboard/profile/3068

### CODE REVIEW

**Review by robin-macpherson — COMMENTED** [review]
Looks great @JoshuaBelden ! I think it would be helpful to include something around the workflow that I gave some notes on in the issue where if someone makes a change in `j-db-client` they will need to re-install the package so that `Prisma Client` is re-generated and up-to-date with those changes.

Other than that it looks great, thanks again for doing this!

**Comment by JoshuaBelden** [comment]
No problem. I'd be happy to add those additions.

**Comment by robin-macpherson** [comment]
@xferguson I know that breaking our Prisma code out into the `j-db-client` package caused some issues for you as well, do you feel like these changes clear everything up or is there anything else you'd like to see in the documentation?

**Comment by JoshuaBelden** [comment]
@robin-macpherson Are you saying when I run `npm i @journaly/j-db-client` it will generate a new migration for any changes to the `schema.prisma` script, replacing the need to run the other to scripts `save` and `up`? When I run `npm i @journaly/j-db-client` I see that it updates the package, but I don't see anything in the console about it creating a new migration.

I'm wondering if I run a `save` and `up` to get my new migration, is there some other steps to generate the prisma client I don't know about, but `npm i @journaly/j-db-client` takes care of that now?

**Comment by robin-macpherson** [comment]
@JoshuaBelden not quite, `PrismaClient` is separate from the actual migrations.

`PrismaClient` is an auto-generated and type-safe query builder that's tailored to our specific data, and we then use that to run all of our database operations in our queries and mutations. The single source of truth for how our data is modeled is the `schema.prisma` file and both the migrations and our `PrismaClient` are generated based on that file.

So when you see us doing `ctx.db.user.findUnique({...})` for example that's all coming from an instance of our `PrismaClient` and gives us nice type safety.

A new client can be generated by running `npm run prisma:generate` and so previously if you made a change to the `schema.prisma` file you'd have to run that to update your local `PrismaClient`, but there is actually a "post-install hook" within the prisma client package that automatically generates the Prisma Client for you, and so when you run `npm install` at the root of the project, that `post-install hook` will generate your updated client.

Does that all make sense?

**Comment by robin-macpherson** [comment]
So the steps are:

1. Update the `schema.prisma` file to make a change to the data model
2. run `npm run migrate:save` to generate the migration for this change
3. run `npm run migrate:up` to apply that migration to your database instance
4. re-install the `@journaly/j-db-client` package in the root project which will also re-generate your `PrismaClient` so that it also reflects the changes to the data model

**Comment by JoshuaBelden** [comment]
That makes a lot of sense and clarifies that I was missing a step. Now I understand why we run `npm run prisma:generate` at the beginning of setup. Looks like that would have been a step we missed discussing later in the doc about migrations, but can now just instruct the user to update the prisma client like you described in the docs.

Thank you.

**Comment by robin-macpherson** [comment]
Awesome! I'm happy that clears things up for you 😄

File: README.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ The data model/DB schema lives in `j-db-client/prisma/schema`, server side code
 
 2. Update your `.env` file with your new postgres username & password. Copy and paste your root `.env` into `j-db-client/prisma/.env`. You should end up with two of the same file, one in root and the other in `j-db-client`.
 
-3. Finally, from the `j-db-client` directory, apply database migrations to your new database instance:
+3. Finally, from the root of the project, apply database migrations to your new database instance:
```
Comment (JoshuaBelden) [inline README.md:62]: I verified that this does in fact work from the root.

File: README.md [hunk 2]
```diff
@@ -76,14 +76,17 @@ The data model/DB schema lives in `j-db-client/prisma/schema`, server side code
...
 ```sh
 $ npm run migrate:save
 $ npm run migrate:up
+$ npm i @journaly/j-db-client
 ```
 
-The first command creates a migration, resulting in a new file in the `j-db-client/prisma/migrations` directory. The second applies that migration to the local database. You'll want to commit any new migration artifacts that you create. Previous migrations should never be edited.
+The first command creates a migration, resulting in a new file in the `j-db-client/prisma/migrations` directory. The second applies that migration to the local database. Finally, re-installing the `@journaly/j-db-client` package results in an updated `PrismaClient`. You'll want to commit any new migration artifacts that you create. 
+
+_Previous migrations should never be edited._
```
Comment (JoshuaBelden) [inline README.md:89]: I thought that this was worth calling out on its own.

File: README.md (hunk 1 above)
Comment (robin-macpherson) [reply README.md:62]: Great!

File: README.md (hunk 2 above)
Comment (robin-macpherson) [reply README.md:89]: Sounds good, I'm always in favor of more clarity.

**Comment by robin-macpherson** [comment]
@JoshuaBelden I'm happy to merge this now, just want to confirm there are no more incoming changes?

**Comment by JoshuaBelden** [comment]
I have nothing left to commit. Thank you for checking.

**Review by robin-macpherson — APPROVED** [review]
Great stuff! 🚀

**Comment by xferguson** [comment]
> @xferguson I know that breaking our Prisma code out into the `j-db-client` package caused some issues for you as well, do you feel like these changes clear everything up or is there anything else you'd like to see in the documentation that would be helpful for new contributors getting started?

Thanks for checking in, @robin-macpherson -- I have been too insanely busy to try and get back on the platform. I'm eager to test this out again soon, and will keep you posted.

**Comment by JoshuaBelden** [comment]
@xferguson If you see any discrepancies or need anything updated I'd be happy to help. Working on documentation is a quick way for me to come up to speed.

Also, nice to meet you. Great app!

**Pull request merged**